### DISCUSSION TITLE
Docker [Alpine]: uWebSockets is incompatible

### DISCUSSION THREAD

**Discussion by dalisoft** [uNetworking/uWebSockets.js#158] [Q&A] [answered]
Hi, @alexhultman 

I trying within Docker with this [node alpine](https://github.com/mhart/alpine-node), but says this library is incompatible.

Reason: `Alpine uses musl instead`, source [here](https://github.com/mhart/alpine-node/issues/140#issuecomment-498783926) and [here](https://github.com/mhart/alpine-node#caveats)

Related issue: [#140](https://github.com/mhart/alpine-node/issues/140)

**Comment by dalisoft** [comment]
I will track this issue [here](https://github.com/dalisoft/nanoexpress/issues/6) and will add solution on README

**Comment by ghost** [comment, marked as answer]
Binaries are not compatible with Alpine since it is using a different libc. You can compile your own binaries if you insist on using this particular embedded dist.

**Reply by dalisoft** [reply]
Sorry @alexhultman, you here wrong, binaries supports Alpine and even CentOS, see my answer below and if you think my answer helpful, you can mark as Answer.

**Comment by dalisoft** [comment]
If anyone also gets this issue, you can try

`RUN ln -s /lib/libc.musl-x86_64.so.1 /lib/ld-linux-x86-64.so.2` command. 
It worked for me. 
uWebSockets.js is amazing, even binaries works with this command. 

If you need solutions for Docker/Linux issues, look at [here](https://nanoexpress.js.org/docker-linux) as my library works on top of uWebSockets.js

**Reply by theoephraim** [reply]
same solution but for multiple architectures
`RUN ln -s "/lib/libc.musl-$(uname -m).so.1" "/lib/ld-linux-$(uname -m).so.1"`

**Reply by kolserdav** [reply]
`uname -m` in Dockerfile always shows `$BUILDPLATFORM` not `$TARGETPLATFORM`

**Comment by kroscek** [comment]
I can run uwebsocket.js inside Docker with additional: RUN apk add --no-cache gcompat

**Reply by kodmanyagha** [reply]
Bro, you saved my 3 days. I want to write a poetry for this answer.

Sometimes I feel that my brain is going to blow.
Searching engines and scrolling to below.
Neither google nor chatgpt nor stackoverflow,
Thanks to all github bros who opens window.

**Comment by theoephraim** [comment]
Had to adjust the fix slightly to work on different architectures.

Adding this to my Dockerfile allowed uws to work: 
`RUN ln -s "/lib/libc.musl-$(uname -m).so.1" "/lib/ld-linux-$(uname -m).so.1"`

**Reply by kolserdav** [reply]
`uname -m` in Dockerfile always shows `$BUILDPLATFORM` not `$TARGETPLATFORM`
